persistence/src/main/scala/hmda/persistence/processing/SubmissionManager.scala

```diff
@@ -151,7 +149,7 @@ object SubmissionManager {
         case SubmissionFSM.name => sender() ! submissionFSM
         case HmdaRawFile.name => sender() ! submissionUpload
         case HmdaFileParser.name => sender() ! submissionParser
-        case HmdaFileValidator.name => sender() ! submissionValidator
+        case HmdaFileValidator.name => submissionValidator.map(sender() ! _)
```

Comment:
This closes over the sender in a future, and can have pretty bad consequences and cause a shared mutable state bug or race conditions. Easy change: 

```scala
case HmdaFileValidator.name =>
  val replyTo = sender()
  submissionValidator.map(client ! _)
```

For more information on this, see this [article](http://helenaedelson.com/?p=879)